MDAnalysis compilation fails on OSX clang 16 [xdrfile]

This is coming out of the conda-forge runners on osx builds:

```
  MDAnalysis/lib/formats/src/xdrfile.c:2535:3: error: incompatible function pointer types initializing 'unsigned int (*)(XDR *)' (aka 'unsigned int (*)(struct XDR *)') with an expression of type 'int64_t (XDR *)' (aka 'long long (struct XDR *)') [-Wincompatible-function-pointer-types]
                  xdrstdio_getpos,                /* get offset in the stream */
                  ^~~~~~~~~~~~~~~
  MDAnalysis/lib/formats/src/xdrfile.c:2536:3: error: incompatible function pointer types initializing 'int (*)(XDR *, unsigned int)' (aka 'int (*)(struct XDR *, unsigned int)') with an expression of type 'int (XDR *, int64_t, int)' (aka 'int (struct XDR *, long long, int)') [-Wincompatible-function-pointer-types]
                  xdrstdio_setpos,                /* set offset in the stream */
                  ^~~~~~~~~~~~~~~
```

--- a/package/MDAnalysis/lib/formats/src/xdrfile.c
+++ b/package/MDAnalysis/lib/formats/src/xdrfile.c
@@ -132,8 +132,8 @@ struct XDR
 		int (*x_getbytes) (XDR *__xdrs, char *__addr, unsigned int __len);
 		int (*x_putbytes) (XDR *__xdrs, char *__addr, unsigned int __len);
 		/* two next routines are not 64-bit IO safe - don't use! */
-		unsigned int (*x_getpostn) (XDR *__xdrs);
-		int (*x_setpostn) (XDR *__xdrs, unsigned int __pos);
+		int64_t (*x_getpostn) (XDR *__xdrs);
+		int (*x_setpostn) (XDR *__xdrs, int64_t __pos, int __whence);
 		void (*x_destroy) (XDR *__xdrs);
 	}
     *x_ops;

